src/app/category-chart/category-chart-samples.module.ts

```diff
@@ -0,0 +1,24 @@
+import { NgModule, ModuleWithProviders  } from '@angular/core';
```

Comment:
I don't think we should necessarily add more modules structure to the samples, any complexity just makes it more confusing for the user to figure out how they need to extract a sample for the samples browser.

Reply:
I initially did this as it's the recommended way to load Feature Modules using services (using ModuleWithProviders and forRoot()).  It prevents multiple instances of the service from being created, since services are globally scoped. If we need to rethink how the whole design will work, that is fine. I think we should wait to hear from Konstantin on the optimization comment below before I go making any more changes.